arithmetization/src/test/java/net/consensys/linea/zktracer/instructionprocessing/callTests/sixtyThreeSixtyFourths/SixtyThreeSixtyFourthsTests.java

```diff
@@ -0,0 +1,390 @@
+    return BytecodeCompiler.newProgram()
+        .push(0) // returnAtCapacity
+        .push(0) // returnAtOffset
+        .push(cds) // callDataSize
+        .push(0) // callDataOffset
+        .push(transfersValue ? 1 : 0) // value
+        .push(address) // address
+        .push(gas) // gas
+        .compile();
+  }
+
+  // TODO: the two methods below are essentially duplicates of the ones in
+  //  LowGasStipendPrecompileCallTests. Consider unifying them in a common class.
+  Bytes storeBlake2fInput(int rLeadingByte) {
+    return BytecodeCompiler.newProgram().push(rLeadingByte).push(2).op(MSTORE8).compile();
+  }
+
+  private Bytes storeModexpInput(int bbs, int mbs, int ebs) {
+    final Bytes32 bbsPadded = Bytes32.leftPad(Bytes.of(bbs));
+    final Bytes32 ebsPadded = Bytes32.leftPad(Bytes.of(ebs));
+    final Bytes32 mbsPadded = Bytes32.leftPad(Bytes.of(mbs));
+    final Bytes bem =
+        Bytes.fromHexString("0x" + "aa".repeat(bbs) + "ff".repeat(ebs) + "bb".repeat(mbs));
+    final Bytes modexpInput = Bytes.concatenate(bbsPadded, ebsPadded, mbsPadded, bem);
+
+    // codeOwnerAccount owns the bytecode that will be given as input to MODEXP through EXTCODECOPY
+    final ToyAccount codeOwnerAccount =
+        ToyAccount.builder()
+            .balance(Wei.of(0))
+            .nonce(1)
+            .address(codeOwnerAddress)
+            .code(modexpInput)
+            .build();
+    additionalAccounts = List.of(codeOwnerAccount);
+
+    // This is computed here for convenience, and it is used for pricing the MODEXP precompile
+    exponentLog =
+        Math.max(OobOperation.computeExponentLog(modexpInput, 96 + bbs + ebs + mbs, bbs, ebs), 1);
+
+    // Copy to targetOffset the code of codeOwnerAccount
+    return BytecodeCompiler.newProgram()
+        .push(codeOwnerAddress)
+        .op(OpCode.EXTCODESIZE) // size
+        .push(0) // offset
+        .push(0) // targetOffset
+        .push(codeOwnerAddress) // address
+        .op(OpCode.EXTCODECOPY)
+        .compile();
+  }
+
+  long getGasLimit(
+      long targetCalleeGas,
+      boolean transfersValue,
+      boolean targetAddressExists,
+      long preCallProgramGas) {
+    /* gasLimit = preCallProgramGasCost + gasPreCall
+    /  63/64 * (gasPreCall - gasUpFront) + stipend = targetCalleeGas
+    /  x = gasPreCall - gasUpFront
+    /  k = x / 64 (integer division)
+    /  l = x - 64 * k
+    /  63 * k + l + stipend = targetCalleeGas
+    / find gasLimit going backwards
+    */
+    final long stipend = transfersValue ? GlobalConstants.GAS_CONST_G_CALL_STIPEND : 0;
+    checkArgument(targetCalleeGas >= stipend);
+    final long l = (targetCalleeGas - stipend) % 63;
+    final long k = (targetCalleeGas - stipend - l) / 63;
+    checkArgument(targetCalleeGas == 63 * k + l + stipend);
+    final long gasUpfront = getUpfrontGasCost(transfersValue, targetAddressExists);
+    final long gasPreCall = (targetCalleeGas - stipend) * 64 / 63 + gasUpfront;
```

Comment:
I think this may be wrong. I would have expected to see
```java
final long gasPreCall = k * 64 + l + gasUpfront;
```
The issue is that integer multiplication and division don't commute, i.e. `(x * p) / q` is not the same as `(x / q) * p`.